### ISSUE TITLE
nltk with sklearn

### ISSUE THREAD

**Issue by btabibian** [nltk/nltk#406]
I saw there is an interface between nltk and sklearn supervised training, or any pipeline that would do the same. I was interested to use sklearn only at the level of feature transformation but the interface does not support it. 

One way to do this is to modify the current solution and make it call "transform" function instead of "predict" function, however, it does not conform with the name of the module "classify".

I am willing to work on this if there is a consensus on how to do this.

**Comment by stevenbird** [comment]
@larsmans do you have any advice please?

**Comment by btabibian** [comment]
for my own work i just changed the current file.

I am not very familiar with nltk, but if there is a module that handles
feature extractions, such as bag of words, etc. then a similar wrapper in
that module would be sensible otherwise modifying current file should be
fine.

**Comment by larsmans** [comment]
I did not implement this because the feature representations of scikit-learn and NLTK differ quite a lot and I wasn't sure if exposing NLTK users to NumPy arrays/SciPy sparse matrices was in order. Besides, nothing in NLTK groks those, I think.

I'd advise just using scikit-learn functionality, since any attempt to wrap all of it in NLTK would just duplicate the API. If the input is in the NLTK format, then [`sklearn.feature_extraction.DictVectorizer`](http://scikit-learn.org/stable/modules/generated/sklearn.feature_extraction.DictVectorizer.html) should be able to handle it.

**Comment by stevenbird** [comment]
@larsmans -- thanks for this.

**Comment by btabibian** [comment]
@larsmans 
Thanks for the input, but why did you implement this in the first place if users could just use DictVectorizer?

Also there is a wide range of tools in sklearn that can be useful for the user, such as regression, semi-supervised learning and feature extraction methods, if an easy way was available to go from nltk to sklearn and come back.

**Comment by larsmans** [comment]
When I wrote the NLTK wrapper, `DictVectorizer` was not yet released. Besides, the API is not 100% the same, as the handling of labels is done differently. I wanted a drop-in replacement for the other NLTK classifiers, so that users can use scikit-learn without changing too much in their code.

I'm not deeply involved in NLTK development, so I don't plan to design and implement new APIs for it.

**Comment by btabibian** [comment]
Thanks for the info,

I actually changed my code to use DictVectorizer.

Best,
Behzad
